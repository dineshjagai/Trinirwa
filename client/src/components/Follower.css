.followerContainer {
  width: 45%;
  height: 200px;
  object-fit: cover;
  border: 1px solid #80cbc4;
  background-color: #D0EAE4;
  border-radius: 5px;
  margin-bottom: 5px;
}

.myimg{
  width: 100%;
  height:100%;
  object-fit:cover;
  border-radius: 5px;
}

.text{
  margin: auto;
  text-align: center;
}

.myimgContainer{
  margin: auto;
  width: 90%;
  height:90%;
}