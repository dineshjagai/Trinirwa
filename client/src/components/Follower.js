import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import BlockIcon from '@material-ui/icons/Block';
import PersonAddIcon from '@material-ui/icons/PersonAdd';
import Tooltip from '@material-ui/core/Tooltip';
import './Follower.css';

const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(2),
      borderRadius: '100%',
      padding: 0,
      overflow: 'none',
    },
  },
}));

export default function Follower({ info, handleB, handleF }) {
  const classes = useStyles();
  return (
    <div className="Container_two">
      <div className={classes.root}>
        <Tooltip title="Block user" placement="top">
          <BlockIcon className="button" color="secondary" onClick={() => handleB(info.username)} />
        </Tooltip>
        <Tooltip title="Follow" placement="top">
          <PersonAddIcon className="button" onClick={() => handleF(info.username)} />
        </Tooltip>
      </div>
      <div className="followerContainer">
        <div className="myimgContainer">
          <img
            className="myimg"
            src={`/api/viewFile/${info.profile_picture}`}
            alt=""
          />
        </div>
      </div>
      <div className="text">{info.username}</div>
    </div>
  );
}
