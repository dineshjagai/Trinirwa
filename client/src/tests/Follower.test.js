import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Follower from '../components/Follower';

test('Test Follower snapshot', () => {
  const i = {
      'username': 'isimbib',
      'profile_picture' : 'logo512.png',
  };
  const component = renderer.create(<Follower info={i}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
