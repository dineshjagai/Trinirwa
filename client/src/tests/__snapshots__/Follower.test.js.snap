// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Follower snapshot 1`] = `
<div
  className="Container_two"
>
  <div
    className="makeStyles-root-1"
  >
    <svg
      aria-describedby={null}
      aria-hidden={true}
      className="MuiSvgIcon-root button MuiSvgIcon-colorSecondary"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Block user"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM4 12c0-4.42 3.58-8 8-8 1.85 0 3.55.63 4.9 1.69L5.69 16.9C4.63 15.55 4 13.85 4 12zm8 8c-1.85 0-3.55-.63-4.9-1.69L18.31 7.1C19.37 8.45 20 10.15 20 12c0 4.42-3.58 8-8 8z"
      />
    </svg>
    <svg
      aria-describedby={null}
      aria-hidden={true}
      className="MuiSvgIcon-root button"
      focusable="false"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Follow"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="followerContainer"
  >
    <div
      className="myimgContainer"
    >
      <img
        alt=""
        className="myimg"
        src="/viewFile/logo512.png"
      />
    </div>
  </div>
  <div
    className="text"
  >
    isimbib
  </div>
</div>
`;
